processor/src/debug.rs

```diff
@@ -0,0 +1,61 @@
+use crate::{ExecutionError, Felt, Process};
+use core::ops::RangeInclusive;
+use vm_core::Word;
+
+/// VmState holds a current process state information at a specific clock cycle.
+#[derive(Clone, Debug)]
+pub struct VmState {
+    pub clk: usize,
+    pub fmp: Felt,
+    pub stack: Vec<Felt>,
+    pub memory: Vec<(u64, Word)>,
+}
+
+/// Iterator that iterates through vm state at each step of the execution.
+/// This allows debuging or replaying ability to view various process state
+/// at each clock cycle.
+/// If the execution returned an error, it returns that error on the clock cycle
+/// it stopped.
+pub struct VmStateIterator {
+    process: Process,
+    error: Option<ExecutionError>,
+    current_clk: usize,
```

Comment:
nit: should `current_clk` be just `clk`?

Reply:
fixed